src/main/java/io/cdap/plugin/gcp/bigquery/sink/BigQuerySinkUtils.java

```diff
@@ -48,12 +58,113 @@
   private static final String temporaryBucketFormat = gcsPathFormat + "/input/%s-%s";
   private static final String DATETIME = "DATETIME";
 
-  public static void configureBigQueryOutput(Configuration configuration,
-                                             String projectName,
-                                             String datasetName,
-                                             String tableName,
-                                             String temporaryGCSPath,
-                                             List<BigQueryTableFieldSchema> fields) throws IOException {
+  /**
+   * Creates Hadoop configuration for the given table and its fields.
+   *
+   * @param bucket bucket name
+   * @param tableName table name
+   * @param fields list of Big Query fields
+   * @return Hadoop configuration
+   */
+  public static Configuration getOutputConfiguration(Configuration baseConfiguration,
```

Comment:
Please make this method name follow same naming (configure*Output)

Reply:
I'd argue that they are different because this method builds a new configuration instance (by copying the supplied one and adding to it), but the other one operates on the supplied configuration by modifying the underlying config instance.

Reply:
Did additional refactor/cleanup. Take a look now.